src/script/scriptBuilder.zig

```diff
@@ -0,0 +1,96 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const Script = @import("lib.zig").Script;
+const StackError = @import("stack.zig").StackError;
+pub const engine = @import("engine.zig");
+const Opcode = @import("./opcodes/constant.zig").Opcode;
+const isUnnamedPushNDataOpcode = @import("opcodes/constant.zig").isUnnamedPushNDataOpcode;
+const ScriptNum = @import("lib.zig").ScriptNum;
+const asBool = Script.asBool;
+const asInt = Script.asInt;
+const testing = std.testing;
+
+/// Maximum script length in bytes
+const MAX_SCRIPT_SIZE = 10000;
+
+///ScriptBuilder is a library to generate easier scripts, useful for faster testing
+pub const ScriptBuilder = struct {
+    /// Number of opcodes in the script(this is not a necessary field and may be removed later)
+    size: ?usize,
+
+    /// Dynamic array holding the opcodes
+    script: std.ArrayList(u8),
+
+    ///Memory Allocator
+    allocator: Allocator,
+
+    /// Initialize a new ScriptBuilder
+    ///
+    /// # Arguments
+    /// - `allocator`: Memory allocator for managing engine resources
+    ///
+    /// # Returns
+    /// - `!ScriptBuilder`: use with `try`
+    pub fn new(allocator: Allocator, opcodeCount: ?u8) !ScriptBuilder {
```

Comment:
`opcodeCount: ?u8`
Those are not opcodes. A script can contain opcode sure, but also raw data such as pubkey or signatures.

Reply:
rename it `capacity: usize`